test/unit/test_log.py

```diff
@@ -15,6 +15,8 @@
 # You should have received a copy of the GNU General Public License
 # along with Autosubmit.  If not, see <http://www.gnu.org/licenses/>.
 
+import pytest
+import pytest_mock
```

Comment:
```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from pytest_mock import MockFixture
```

Then in the test signature `mocker : 'MockFixture'`. This way we only keep imports of code used in the code (avoids actually importing the code just for types which may slow down code (not important in a test file to be honest) and avoids issues with circular dependencies.